lib/fetch.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const nodeFetch = require('node-fetch');
+const { omitBy, isUndefined } = require('lodash');
+
+const omitUndefined = object => omitBy(object, isUndefined);
+
+const errorWith = async (error, url, options, response, responseText) => Object.assign(error, {
+    meta: omitUndefined({
+        request: omitUndefined({ url, method: options.method || 'GET', body: options.body }),
+        response: response && {
+            status: response.status,
+            body: responseText
+        }
+    })
+});
+
+/**
+ * @param {string} url
+ * @param {{
+ *     expectOk?: boolean
+ *     parseResponseAs?: 'json'|'text'
+ * }} [options] - extended Fetch options
+ * @returns {Promise.<Response|string|*>} - Returns Response object or parsed body value if parseResponseAs is specified
```

Comment:
I would rather use direct reference to "Response", stated as "nodeFetch.Response", because it points out to correct class and IDE can then actually provide with better context help.